app/src/main/java/org/thoughtcrime/securesms/groups/compose/SelectContactsScreen.kt

```diff
@@ -0,0 +1,144 @@
+import org.session.libsignal.utilities.AccountId
+import org.thoughtcrime.securesms.groups.ContactItem
+import org.thoughtcrime.securesms.groups.SelectContactsViewModel
+import org.thoughtcrime.securesms.ui.SearchBar
+import org.thoughtcrime.securesms.ui.components.BackAppBar
+import org.thoughtcrime.securesms.ui.components.PrimaryOutlineButton
+import org.thoughtcrime.securesms.ui.theme.LocalColors
+import org.thoughtcrime.securesms.ui.theme.PreviewTheme
+
+
+@Serializable
+object RouteSelectContacts
+
+@Composable
+fun SelectContactsScreen(
+    excludingAccountIDs: Set<AccountId> = emptySet(),
+    onDoneClicked: (selectedContacts: Set<AccountId>) -> Unit,
+    onBackClicked: () -> Unit,
+) {
+    val viewModel = hiltViewModel<SelectContactsViewModel, SelectContactsViewModel.Factory> { factory ->
+        factory.create(excludingAccountIDs)
+    }
+
+    SelectContacts(
+        contacts = viewModel.contacts.collectAsState().value,
+        onContactItemClicked = viewModel::onContactItemClicked,
+        searchQuery = viewModel.searchQuery.collectAsState().value,
+        onSearchQueryChanged = viewModel::onSearchQueryChanged,
+        onDoneClicked = { onDoneClicked(viewModel.currentSelected) },
+        onBack = onBackClicked,
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SelectContacts(
+    contacts: List<ContactItem>,
+    onContactItemClicked: (accountId: AccountId) -> Unit,
+    searchQuery: String,
+    onSearchQueryChanged: (String) -> Unit,
+    onDoneClicked: () -> Unit,
+    onBack: () -> Unit,
+    @StringRes okButtonResId: Int = R.string.ok
+) {
+    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        BackAppBar(
+            title = stringResource(id = R.string.contactSelect),
+            onBack = onBack,
+        )
+
+        GroupMinimumVersionBanner()
+        SearchBar(
+            query = searchQuery,
+            onValueChanged = onSearchQueryChanged,
+            placeholder = stringResource(R.string.searchContacts),
+            modifier = Modifier.padding(horizontal = 16.dp),
+            backgroundColor = LocalColors.current.backgroundSecondary,
+        )
+
+        LazyColumn(modifier = Modifier.weight(1f)) {
+            multiSelectMemberList(
+                contacts = contacts,
+                onContactItemClicked = onContactItemClicked,
+            )
+        }
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(
+                    verticalGradient(
+                        0f to Color.Transparent,
+                        0.2f to LocalColors.current.background,
+                    )
+                )
+        ) {
+            PrimaryOutlineButton(
+                onClick = onDoneClicked,
+                modifier = Modifier
+                    .padding(horizontal = 12.dp, vertical = 4.dp)
+                    .defaultMinSize(minWidth = 128.dp),
```

Comment:
This uses a different value from the other screen (which was set to 120.dp). Figma seems to use 160.dp. This value should be reused and come from LocalDimensions.
Also the padding above seems wrong. I don't think the horizontal is required.
The bottom should be set to `xsSpacing`
As for the top it should be `spacing` but only as part of the container which should be represented as a fading gradient, which I don't see as implemented here, instead we have a solid background.
I know someone implemented `.fadingEdges` but it's unfortunately tied to Columns' scroll, and also doesn't cater to only bottom fading edges, so we might need to create out own version for this, as this will likely be a reused component.

Reply:
Done